include/zephyr/net/coap.h

```diff
@@ -989,14 +1002,15 @@ void coap_reply_init(struct coap_reply *reply,
  * confirmation message, initialized with data from @a request
  * @param request Message waiting for confirmation
  * @param addr Address to send the retransmission
- * @param retries Maximum number of retransmissions of the message.
+ * @param params Pointer to the CoAP transmission parameters struct,
+ * or NULL to use default values set by Kconfig
```

Comment:
nit: I'd drop the `set by Kconfig` part, as it's not necessarily correct - the initial Kconfig values could be overwritten by `coap_set_transmission_parameters()`.